source/ota.c

```diff
@@ -2128,9 +2142,8 @@ static void handleSelfTestJobDoc( OtaFileContext_t * pFileContext )
     /* Validate version of the update received.*/
     errVersionCheck = validateUpdateVersion( pFileContext );
 
-    /* MISRA rule 14.3 requires controlling expressions to be not invariant. otaconfigAllowDowngrade is
-     * one of the OTA library configuration and it's set to 0 when running the static analysis. But
-     * users can change it when they build their application. So this is a false positive. */
+    /* MISRA Ref 14.3.1 [Configuration dependent invariant] */
+    /* More details at: https://github.com/aws/ota-for-aws-iot-embedded-sdk/blob/main/MISRA.md#rule-143 */
     /* coverity[misra_c_2012_rule_14_3_violation] */
     if( ( otaconfigAllowDowngrade == 1U ) || ( errVersionCheck == OtaErrNone ) )
```

Comment:
is otaconfigAllowDowngrade a static config ? So can we use the pre-processor check instead?
```
#if( otaconfigAllowDowngrade == 1 ) 
```